### PR TITLE
test: Display skipped tests reason

### LINKED ISSUE #26023: Running functional tests using test_runner.py does NOT display the SkipTest() message
**Is your feature request related to a problem? Please describe.**
 When I'm running functional tests using test_runner.py, I am not seeing. SkipTest messages being displayed. For example:
 
_2/254 - feature_bind_extra.py skipped_

Whereas in feature_bind_extra.py, I see this following message:
_raise SkipTest("This test can only be run on Linux.")_

**Describe the solution you'd like**
1. Am I missing a particular configuration when running test_runner.py? Currently, I'm running this: 
_test/functional/test_runner.py --extended -j 10_

2.  If the feature is missing to display SkipTests, would it be a good idea to add this feature to display why tests are getting skipped when using test_runner.py?

### PR DESCRIPTION
**Pull request by aureleoules**
Attempt to fix #26023.

### CODE REVIEW

**Review by kouloumos — COMMENTED** [review]
ACK f5a46ab53413679d7f3514c816ef7d893f0019da, looks good and it definitely adds more context to someone running the functional tests to better understand why those tests are skipped.

Note: As most of the `skip_test_if_missing_module` checks are based on `./configuration` options, reviewers can check for different skip reasons just with different configuration parameters instead of re-building.

**Comment by brunoerg** [comment]
Concept ACK

File: test/functional/test_runner.py [hunk 1]
```diff
@@ -555,14 +555,14 @@ def run_tests(*, test_list, src_dir, build_dir, tmpdir, jobs=1, enable_coverage=
...
-        for test_result, testdir, stdout, stderr in job_queue.get_next():
+        for test_result, testdir, stdout, stderr, reason in job_queue.get_next():
             test_results.append(test_result)
             i += 1
             done_str = "{}/{} - {}{}{}".format(i, test_count, BOLD[1], test_result.name, BOLD[0])
             if test_result.status == "Passed":
                 logging.debug("%s passed, Duration: %s s" % (done_str, test_result.time))
             elif test_result.status == "Skipped":
-                logging.debug("%s skipped" % (done_str))
+                logging.debug("%s skipped (%s)" % (done_str, reason))
```
Comment (MarcoFalke) [inline test/functional/test_runner.py:565]:
```suggestion
                logging.debug(f"{done_str} skipped ({reason})")
```

nit: Could use f-strings? Also, could name this `skip_reason`?

**Review by maflcko — APPROVED** [review]

File: test/functional/test_runner.py (hunk 1 above)
Comment (aureleoules) [reply test/functional/test_runner.py]: pushed

**Review by brunoerg — APPROVED** [review]
ACK 07b6e743147f410d8ceb4b6b3282e9fba4cafe50

**Pull request merged**